### ISSUE TITLE
Clarification on Training Effort used to create Qwen-2.5-7B-SimpleRL-Zoo

### ISSUE THREAD

**Issue by cheongalc** [hkust-nlp/simpleRL-reason#78]
Dear authors,

I am currently working on fine-tuning the Qwen-2.5-7B base model using the `simplelr_qwen_level3to5` dataset, aiming to replicate the training conditions of your `Qwen-2.5-7B-SimpleRL-Zoo` model as closely as possible. However, my setup differs in that I am doing supervised fine-tuning on the gold CoTs of `simplelr_qwen_level3to5`, rather than Zero-style reinforcement learning (RL), so not all the hyperparameters transfer directly. 

My goal is to train a model on the same data you used, and with a comparable token exposure, but my training strategy has changed to supervised finetuning.

I would like to confirm a few points to better approximate the training effort involved. I note that the README states

> To accelerate our experiments, we utilized 2x8 H100-80G GPUs to train the 7B and 14B models for approximately 100 steps over 15 hours using 8K examples.

Additionally, in issue #75, it was clarified that the "100 steps" refers to the total number of rollouts, not gradient steps, therefore the number of epochs can be calculated as:

```
8532 data size / 1024 rollout batch size ≈ 8.33 rollouts per epoch  
Thus, 100 rollouts / 8.33 rollouts per epoch ≈ 12 epochs.
```

Furthermore, the paper on page 4 mentions generating 8 rollouts per prompt, and the README confirms that for Qwen-2.5-7B, the train command includes `--rollout_n 8`.

To ensure alignment, I would like to confirm the following:

- Does "rollout batch size of 1024" means that at each training step, 1024 prompts are processed in parallel?
- Does `--rollout_n 8` imply that for each of those 1024 prompts, the model generates 8 responses before rewards are computed and the policy is updated?
- Since this process was repeated for 100 rollouts, totaling approx 12 epochs over the full dataset, does it mean that the model has seen each of the 8,532 prompts roughly 8 x 12 = 96 times? 
	- If so, then in order to match this training effort with supervised fine-tuning, I would need to train for 96 epochs, because each prompt is seen once per epoch.
    
- I obtained the train data of `simplelr_qwen_level3to5` from [here](https://huggingface.co/datasets/hkust-nlp/SimpleRL-Zoo-Data/tree/main/simplelr_qwen_level3to5). May I know if the `test.parquet` in that directory refers to the MATH500? Also, in the paper page 4, it was stated that an additional 500 problems were uniformly sampled for validation. Is the validation data stored in a separate dataset file, or was it sampled from `train.parquet`, so the actual train samples is around 8000, with 500 going to validation?
    
- Finally, may I know if the model released as Qwen-2.5-7B-SimpleRL-Zoo on Hugging Face is the result of this 100-rollout training process?

Thank you so much for your time. I appreciate any clarification you can provide!

**Comment by Zeng-WH** [comment]
Thanks! Hope the following information helps you!

**A1:** Yes, each training step consists of 1024 prompts.  

**A2:** Yes, the model generates 8 responses.  

**A3:** Your calculation of the times is correct.  

**A4:** Yes, `test.parquet` is MATH500. We did not include these validation data in the released dataset.  

**A5:** The released **Qwen-2.5-7B-SimpleRL-Zoo** is a **90-rollout training model**.

**Comment by cheongalc** [comment]
Thank you very much!!
